### PR TITLE
Count of zero sum subarrays c++

### PR DESCRIPTION
**Pull request by gokulkannan7**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added code for the count of zero sum subarray problem with 2 methods.
Added link to README.md in the c++ folder

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2322

### CODE REVIEW

File: C-Plus-Plus/other/Count_Zero_Subarrays.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    cin >> n;
+    ll a[n];
+    for (int i = 0; i < n; i++)
+    {
+        cin >> a[i];
+    }
+    cout << Count_subarrays_brute(a, n) << endl;     // Brute force
+    cout << Count_Subarrays_Optimised(a, n) << endl; // Optimised
+    return 0;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Count_Zero_Subarrays.cpp:77]: Add a new line at the EOF

File: C-Plus-Plus/other/Count_Zero_Subarrays.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#include <bits/stdc++.h>
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Count_Zero_Subarrays.cpp:1]: Add a beginner friendly pitch about your Code/Agorithm

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -187,8 +187,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)]
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Revert these changes

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by gokulkannan7** [comment]
@HarshCasper, I have made the changes, can u review it?

**Comment by gokulkannan7** [comment]
@HarshCasper, can you review the code once, is there anything to be changed?

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "C_Cpp.errorSquiggles": "Disabled"
+}
```
Comment (HarshCasper) [inline .vscode/settings.json:3]: Remove this

File: C-Plus-Plus/other/Count_Zero_Subarrays.cpp (hunk 2 above)
Comment (HarshCasper) [reply C-Plus-Plus/other/Count_Zero_Subarrays.cpp:1]: Not yet resolved

File: C-Plus-Plus/other/Count_Zero_Subarrays.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+Output Format
+Find the count of the total zero-sum subarrays.
+
+Sample Input
+11
+6 3 -1 -3 4 -2 2 4 6 -12 -7
+
+Sample Output
+5
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Count_Zero_Subarrays.cpp:20]: Add sample I/O at the end of the Code

File: C-Plus-Plus/other/Count_Zero_Subarrays.cpp (hunk 1 above)
Comment (HarshCasper) [reply C-Plus-Plus/other/Count_Zero_Subarrays.cpp:77]: Not yet resolved

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
Closing due to inactivity.

**Pull request closed without merging**